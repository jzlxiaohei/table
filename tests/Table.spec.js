import React from 'react';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Table from '..';

describe('Table', () => {
  it('renders correctly', () => {
    const columns = [
      { title: '表头1', dataIndex: 'a', key: 'a', colSpan: 2,
        width: 100, render(o, row, index) {
          // 第一列中第一行合并两列
          const obj = {
            children: o,
            props: {},
          };
          if (index === 0) {
            obj.props.colSpan = 2;
          }
          return obj;
        } },
      { id: '123', title: '表头2', colSpan: 0, dataIndex: 'b', key: 'b',
        width: 100, render(o, row, index) {
          // 2列被合并掉了colSpan:0，第二列中第一行合并两行rowSpan:2
          const obj = {
            children: o,
            props: {},
          };
          if (index === 0) {
            obj.props.colSpan = 0;
          }
          return obj;
        } },
      { title: '表头3', dataIndex: 'c', key: 'c', width: 200, render(o, row, index) {
        const obj = {
          children: o,
          props: {},
        };
        if (index === 0) {
          obj.props.rowSpan = 2;
        }
        if (index === 1) {
          obj.props.rowSpan = 0;
        }
        return obj;
      } },
      {
        title: 'operation', dataIndex: '', key: 'operation', render() {
          return <a href="#">操作</a>;
        },
      },
      {
        title: 'number', dataIndex: '', key: 'number', render() {
          return 123;
        },
      },
      {
        title: 'zero', dataIndex: '', key: 'zero', render() {
          return 0;
        },
      },
      {
        title: 'empty string', dataIndex: '', key: 'empty-string', render() {
          return '';
        },
      },
      {
        title: 'string', dataIndex: '', key: 'string', render() {
          return 'text';
        },
      },
      {
        title: 'false', dataIndex: '', key: 'false', render() {
          return false;
        },
      },
      {
        title: 'Array', dataIndex: '', key: 'array', render() {
          return [<a href="#" key="1">操作1</a>, <a href="#" key="2">操作2</a>];
        },
      },
    ];
    const data = [
      { a: '123', key: '1' },
      { a: 'cdd', b: 'edd', key: '2' },
      { a: '1333', c: 'eee', d: 2, key: '3' },
      { a: {}, key: '4' },
    ];
    const wrapper = render(
      <Table
        columns={columns}
        data={data}
        expandedRowRender={record => <p>{record.a}</p>}
        expandRowByClick
      />
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });
});

