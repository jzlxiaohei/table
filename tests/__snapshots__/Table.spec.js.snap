exports[`Table renders correctly 1`] = `
<div
  class="rc-table rc-table-scroll-position-left">
  <div
    class="rc-table-content">
    <div
      class="">
      <span>
        <div
          class="rc-table-body">
          <table
            class="">
            <colgroup>
              <col
                style="width:100px;min-width:100px;" />
              <col
                style="width:100px;min-width:100px;" />
              <col
                style="width:200px;min-width:200px;" />
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
            </colgroup>
            <thead
              class="rc-table-thead">
              <tr>
                <th
                  class=""
                  colspan="2">
                  表头1
                </th>
                <th
                  class="">
                  表头3
                </th>
                <th
                  class="">
                  operation
                </th>
                <th
                  class="">
                  number
                </th>
                <th
                  class="">
                  zero
                </th>
                <th
                  class="">
                  empty string
                </th>
                <th
                  class="">
                  string
                </th>
                <th
                  class="">
                  false
                </th>
                <th
                  class="">
                  Array
                </th>
              </tr>
            </thead>
            <tbody
              class="rc-table-tbody">
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class=""
                  colspan="2">
                  123
                </td>
                <td
                  class=""
                  rowspan="2" />
                <td
                  class="">
                  <a
                    href="#">
                    操作
                  </a>
                </td>
                <td
                  class="">
                  123
                </td>
                <td
                  class="">
                  0
                </td>
                <td
                  class="" />
                <td
                  class="">
                  text
                </td>
                <td
                  class="" />
                <td
                  class="">
                  <a
                    href="#">
                    操作1
                  </a>
                  <a
                    href="#">
                    操作2
                  </a>
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="">
                  cdd
                </td>
                <td
                  class="">
                  edd
                </td>
                <td
                  class="">
                  <a
                    href="#">
                    操作
                  </a>
                </td>
                <td
                  class="">
                  123
                </td>
                <td
                  class="">
                  0
                </td>
                <td
                  class="" />
                <td
                  class="">
                  text
                </td>
                <td
                  class="" />
                <td
                  class="">
                  <a
                    href="#">
                    操作1
                  </a>
                  <a
                    href="#">
                    操作2
                  </a>
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="">
                  1333
                </td>
                <td
                  class="" />
                <td
                  class="">
                  eee
                </td>
                <td
                  class="">
                  <a
                    href="#">
                    操作
                  </a>
                </td>
                <td
                  class="">
                  123
                </td>
                <td
                  class="">
                  0
                </td>
                <td
                  class="" />
                <td
                  class="">
                  text
                </td>
                <td
                  class="" />
                <td
                  class="">
                  <a
                    href="#">
                    操作1
                  </a>
                  <a
                    href="#">
                    操作2
                  </a>
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="" />
                <td
                  class="" />
                <td
                  class="" />
                <td
                  class="">
                  <a
                    href="#">
                    操作
                  </a>
                </td>
                <td
                  class="">
                  123
                </td>
                <td
                  class="">
                  0
                </td>
                <td
                  class="" />
                <td
                  class="">
                  text
                </td>
                <td
                  class="" />
                <td
                  class="">
                  <a
                    href="#">
                    操作1
                  </a>
                  <a
                    href="#">
                    操作2
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </span>
    </div>
  </div>
</div>
`;
